File: app/src/main/java/com/ndmsystems/coala/observer/Observer.java
package com.ndmsystems.coala.observer;

import com.ndmsystems.coala.helpers.Hex;
import com.ndmsystems.coala.message.CoAPMessage;

import java.net.InetSocketAddress;

/**
 * Created by bas on 14.11.16.
 */

public class Observer {
    public final CoAPMessage registerMessage;
    public final InetSocketAddress address;

    public Observer(CoAPMessage registerMessage, InetSocketAddress address) {
        this.registerMessage = registerMessage;
        this.address = address;
    }

    @Override
    public boolean equals(Object otherObject) {
        if (this == otherObject)
            return true;
        if (otherObject == null)
            return false;
        if (getClass() != otherObject.getClass())
            return false;
        Observer other = (Observer) otherObject;
        return key().equals(other.key());
    }

    public String key() {
        return Hex.encodeHexString(registerMessage.getToken());
    }

    @Override
    public String toString() {
        return "Uri: " + registerMessage.getURI()
                + " token: " + registerMessage.getHexToken()
                + " address: " + (address != null ? address.toString() : null);
    }
}

File: app/src/main/java/com/ndmsystems/coala/observer/Observer.kt
package com.ndmsystems.coala.observer

import com.ndmsystems.coala.helpers.Hex.encodeHexString
import com.ndmsystems.coala.message.CoAPMessage
import java.net.InetSocketAddress

/**
 * Created by bas on 14.11.16.
 */
class Observer(val registerMessage: CoAPMessage, val address: InetSocketAddress?) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null) return false
        if (javaClass != other.javaClass) return false
        val other = other as Observer
        return key() == other.key()
    }

    fun key(): String {
        return encodeHexString(registerMessage.token)
    }

    override fun toString(): String {
        return ("Uri: " + registerMessage.getURI()
                + " token: " + registerMessage.hexToken
                + " address: " + address?.toString())
    }
}